src/AppInstallerCLICore/Commands/COMInstallCommand.cpp

```diff
@@ -13,12 +14,25 @@ using namespace AppInstaller::Utility::literals;
 namespace AppInstaller::CLI
 {
     // IMPORTANT: To use this command, the caller should have already retrieved the package manifest (GetManifest()) and added it to the Context Data
-    void COMInstallCommand::ExecuteInternal(Context& context) const
+    void COMDownloadCommand::ExecuteInternal(Context& context) const
     {
+        // TODO: Remove duplication with InstallFlow
         context <<
             Workflow::ReportExecutionStage(ExecutionStage::Discovery) <<
             Workflow::SelectInstaller <<
             Workflow::EnsureApplicableInstaller <<
-            Workflow::InstallSinglePackage;
+            Workflow::ReportIdentityAndInstallationDisclaimer <<
+            Workflow::ShowPackageAgreements(/* ensureAcceptance */ true) <<
+            Workflow::GetDependenciesFromInstaller <<
+            Workflow::ReportDependencies(Resource::String::InstallAndUpgradeCommandsReportDependencies) <<
+            Workflow::ReportExecutionStage(ExecutionStage::Download) <<
+            Workflow::DownloadInstaller;
+    }
+
+    // IMPORTANT: To use this command, the caller should have already executed the COMDownloadCommand
+    void COMInstallCommand::ExecuteInternal(Context& context) const
+    {
+        context <<
+            Workflow::InstallPackageInstaller;
```

Comment:
>InstallPackageInstaller [](http://example.com/codeflow?start=22&length=23)

I see that this still tries to do a download in cases where the file no longer exists or the hash is mismatched. I think that will probably not be the best idea once simultaneous downloads comes in since part of the goal of that feature is to not have installs need to wait for other downloads to finish.

Reply:
I agree that in this case we don't want to perform the download on the second command; just verify the file and proceed to install.